src/schoology-extractor/edfi_schoology_extractor/api/request_client.py

```diff
@@ -184,7 +184,7 @@ def _check_response(
         self._check_for_success(response, success_status)
 
     @retry(
-        retry_on_exceptions=(ConnectionError, HTTPError, ProtocolError, Timeout),
+        retry_on_exceptions=(ConnectionError, HTTPError, ProtocolError, Timeout, RuntimeError),
```

Comment:
I wonder about this. I have seen 502 from Schoology before, and I tend to think that complete failure is a good thing: while the service will come up at some point, a 502 probably means they're having severe problems and thus we might want to stop our requests.

If we *don't* handle it here, then we might want to document. Overall, I had not thought about documenting the retry functionality, but for end users this might be very valuable.

Reply:
Yeah, I'm not 100% on this mostly because "RuntimeError" is such a blanket exception. However, I got 502's several times while testing and re-running immediately went fine, which leads me to think it's just some sporadic issue.  Our default retry doesn't really retry too hard.  With time-limited exponential backoff, I think we retry four times in one minute before giving up.

Reply:
If it fails four times, the whole script will bomb out, right? Rather than moving on to the next resource? If so that seems reasonable to me.

Reply:
That's correct.  It rethrows the exception at that point.